client/client.cpp

```diff
@@ -0,0 +1,168 @@
+        outerVerticalLayout->addItem(new QSpacerItem(0, 0, QSizePolicy::Ignored, QSizePolicy::MinimumExpanding));
+        auto outerHorizontalLayout = new QHBoxLayout;
+        outerHorizontalLayout->addItem(new QSpacerItem(0, 0, QSizePolicy::MinimumExpanding, QSizePolicy::Ignored));
+        auto groupBox = new QGroupBox(QGuiApplication::applicationDisplayName());
+        mainLayout = new QGridLayout(groupBox);
+        outerHorizontalLayout->addWidget(groupBox);
+        outerHorizontalLayout->addItem(new QSpacerItem(0, 0, QSizePolicy::MinimumExpanding, QSizePolicy::Ignored));
+        outerVerticalLayout->addLayout(outerHorizontalLayout);
+        outerVerticalLayout->addItem(new QSpacerItem(0, 0, QSizePolicy::Ignored, QSizePolicy::MinimumExpanding));
+    }
+    else
+    {
+        mainLayout = new QGridLayout(this);
+    }
+    mainLayout->addWidget(hostLabel, 0, 0);
+    mainLayout->addWidget(hostCombo, 0, 1);
+    mainLayout->addWidget(portLabel, 1, 0);
+    mainLayout->addWidget(portLineEdit, 1, 1);
+    mainLayout->addWidget(statusLabel, 2, 0, 1, 2);
+    mainLayout->addWidget(buttonBox, 3, 0, 1, 2);
+    setWindowTitle(QGuiApplication::applicationDisplayName());
+    portLineEdit->setFocus();
+
+}
+void Client::requestNew()
+{
+    getButton->setEnabled(false);
+    tcpSocket->abort();
+
+    tcpSocket->connectToHost(hostCombo->currentText(),
+                             portLineEdit->text().toInt());
+}
+
+void Client::displayError(QAbstractSocket::SocketError socketError)
+{
+    switch (socketError) {
+    case QAbstractSocket::RemoteHostClosedError:
+        break;
+    case QAbstractSocket::HostNotFoundError:
+        QMessageBox::information(this, tr(" Client"),
+                                 tr("The host was not found. Please check the "
+                                    "host name and port settings."));
+        break;
+    case QAbstractSocket::ConnectionRefusedError:
+        QMessageBox::information(this, tr(" Client"),
+                                 tr("The connection was refused by the peer. "
+                                    "Make sure the fortune server is running, "
+                                    "and check that the host name and port "
+                                    "settings are correct."));
+        break;
+    default:
+        QMessageBox::information(this, tr(" Client"),
+                                 tr("The following error occurred: %1.")
+                                 .arg(tcpSocket->errorString()));
+    }
+
+    getButton->setEnabled(true);
+}
+void Client::read()
+{
+    in.startTransaction();
+
+
+    in >> list;
+    qDebug() << list;
+
+    QFile* destinationFile= new QFile(QStringLiteral("imageFile.jpg"));
+    Q_ASSUME(destinationFile->open(QFile::WriteOnly));
+   for(int i = 0; i < list.size();++i)
+   {
+    QString l = list[i];
+    QUrl url = QUrl::fromUserInput( l);
+
+
+
+    QNetworkAccessManager* manager = new QNetworkAccessManager(this);
```

Comment:
you create new network managers per each read function call, which doesn't make sense, better to init it in constructor or on tcp socket setup.